src/App.js

```diff
@@ -1,25 +1,63 @@
-import logo from './logo.svg';
+import React, {useState} from 'react'
 import './App.css';
+import foodsJSON from "./foods.json";
+import FoodBox from './components/FoodBox';
+import { Card, Row, Col, Divider, Input, Button } from "antd";
+import AddFoodForm from './components/AddFoodForm';
+import Search from './components/Search';
+
 
 function App() {
+
+ const [ allFoods, setAllFoods ] = useState(foodsJSON)
+ const [ allSearchFood, setAllSearchFood ] = useState(foodsJSON)
+
+
+ const createFood = (food) => {
+  const newFood = [food, ...allFoods]
+  setAllFoods(newFood)
+ }
+
+ const searchFood = (query) => {
+  const filteredFoods = allSearchFood.filter((food) =>
+  food.name.toLowerCase().includes(query.toLowerCase())
+  );
+  setAllFoods(filteredFoods);
+};
+
+const deleteFood = (foodId) => {
+  const filteredFoods = allSearchFood.filter((manyFood) => {
+    return manyFood.name !== foodId;
+  });
+  setAllSearchFood(filteredFoods)
```

Comment:
Here, the deleteFood function is not working when `setAllSearchFood` is called because it uses the `allSearchFood` state to filter out the food to be deleted. The `allSearchFood` state is only updated when the `searchFood` function is called, and it is not updated in the deleteFood function.

This should fix it:
```
const filteredFoods = allFoods.filter((manyFood) => {
  return manyFood.name !== foodId;
});
```